### PR TITLE
[BugFix][Android RPC] fix build bugs and improve android_rpc experience

### PR DESCRIPTION
**Pull request by new-TonyWang**
Fix the following bugs:
1. gradle build fail due to Execution failed for` task ':app:buildJni'. Process 'command 'sh'' finished with non-zero exit value 2`
2. run python android_rpc_test.py and wait for so long but finally got a `Connect to x.x.x.x:5001 `failed
3. RPCActivity may starts up not once but twice

@yzhliu @eqy @tqchen

### CODE REVIEW

**Comment by new-TonyWang** [comment]
There still remains some maven errors in jvm due to the incompatible java version and lack of jni head file paths in 'jvm/native/linux-x86_64', maybe the windows have the same error but I have no windows pc to test it. I think one  way is to require user to set a $JAVA_HOME in path, then we are able to set the jni head file paths easily.

**Review by new-TonyWang — COMMENTED** [review]
ok

File: apps/android_rpc/.gitignore [hunk 1]
```diff
@@ -6,4 +6,7 @@
 .DS_Store
 /build
 /captures
-.externalNativeBuild
+/.idea
+/gradlew
+/gradlew.bat
+/gradle
```
Comment (liangfu) [inline apps/android_rpc/.gitignore:12]: please add newline at EOF

File: apps/android_rpc/app/.gitignore [hunk 2]
```diff
@@ -1 +1,7 @@
 /build
+/.externalNativeBuild
+/src/main/jni/org_apache_tvm_native_c_api.h
+/src/main/jni/org_apache_tvm_native_c_api.cc
+/src/main/jni/jni_helper_func.h
+/src/main/obj
+/libs
```
Comment (liangfu) [inline apps/android_rpc/app/.gitignore:7]: please add newline at EOF

File: apps/android_rpc/app/build.gradle [hunk 3]
```diff
@@ -35,17 +35,31 @@ android {
         versionCode 1
         versionName "1.0"
         testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
+
+        externalNativeBuild{
+            cmake {
+                abiFilters "arm64-v8a", "armeabi-v7a","x86","x86_64"
+            }
+        }
+
```
Comment (liangfu) [inline apps/android_rpc/app/build.gradle:44]: please remove extra blank lines.

File: apps/android_rpc/app/src/main/jni/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,51 @@
+cmake_minimum_required(VERSION 3.4.1)
+
+add_library(
+        tvm4j_runtime_packed
+
```
Comment (liangfu) [inline apps/android_rpc/app/src/main/jni/CMakeLists.txt:5]: please remove extra blank lines, and some similar ones in the following.

File: apps/android_rpc/app/src/main/jni/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    set_target_properties(
+            tvm4j_runtime_packed
+
+            PROPERTIES IMPORTED_LOCATION
+
+            ${CMAKE_SOURCE_DIR}/../obj/local/x86_64/libtvm4j_runtime_packed.so
+
+    )
+
+endif()
```
Comment (liangfu) [inline apps/android_rpc/app/src/main/jni/CMakeLists.txt:51]: please add newline at EOF

File: apps/android_rpc/app/src/main/jni/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+        PROPERTIES IMPORTED_LOCATION
+
+        ${CMAKE_SOURCE_DIR}/../obj/local/x86/libtvm4j_runtime_packed.so
+
+)    
+
+elseif(${ANDROID_ABI} STREQUAL "x86_64")
+    set_target_properties(
+            tvm4j_runtime_packed
```
Comment (liangfu) [inline apps/android_rpc/app/src/main/jni/CMakeLists.txt:43]: please keep indentation consistent.

File: apps/android_rpc/app/build.gradle [hunk 7]
```diff
@@ -35,17 +35,31 @@ android {
...
-            jniLibs.srcDirs = ['src/main/libs']
+    externalNativeBuild{
+        cmake {
+                path "src/main/jni/CMakeLists.txt"
+            }
+    
+        sourceSets {
+            main {
+                jni.srcDirs = []
+                //jniLibs.srcDirs = ['src/main/libs']
```
Comment (liangfu) [inline apps/android_rpc/app/build.gradle:61]: please remove unused code.

File: apps/android_rpc/app/src/main/java/org/apache/tvm/tvmrpc/MainActivity.java [hunk 8]
```diff
@@ -69,6 +69,7 @@ public Intent updateRPCPrefs() {
     editor.putString("input_key", key);
     editor.putBoolean("input_switch", isChecked);
     editor.commit();
+
```
Comment (liangfu) [inline apps/android_rpc/app/src/main/java/org/apache/tvm/tvmrpc/MainActivity.java:72]: remove extra blank lines.

**Review by liangfu — CHANGES_REQUESTED** [review]
Thanks for improving android_rpc.

I left some comments, otherwise looks good.

**Comment by jroesch** [comment]
This PR appears to be out of date, please feel free to reopen it if this is not the case.

As part of the new year we are attempting to triage the project's open pull requests to ensure that code which
is ready for review and/or merging receives adequate attention.

Thanks again for your contribution, and feel free to reach out to discuss these changes.

**Pull request closed without merging**